text/3417-dropck-eyepatch-v3.md

```diff
@@ -0,0 +1,250 @@
+- Feature Name: (`dropck_eyepatch_v3`)
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Cleanup the rules for implicit drops by adding an argument for `#[may_dangle]` on type
+parameters: `#[may_dangle(droppable)]` and `#[may_dangle(must_not_use)]`. Change `PhantomData`
+to get completely ignored by dropck as its current behavior is confusing and inconsistent.
+
+# Motivation
+[motivation]: #motivation
+
+`PhantomData` is `Copy` but still adds outlives requirements when dropped as a part of
+a larger value. This behavior is inconsistent and results in "spooky-dropck-at-a-distance" 
```

Comment:
I personally don't find this phrase ("spooky-dropck-at-a-distance") very clear and would recommend we drop it from the RFC and replace it with something more clearly meaningful. My best effort to understand what "spooky dropck at a distance means" is something like "This behavior means that the interaction of PhantomData and dropck cannot be understood in isolation, but instead has to be considered as part of the larger type that the PhantomData is embedded in". Does that sound right?

(As a stylistic side note, although I sometimes forget this myself, I generally find it's better to avoid "cutesy" negative labels for previous systems. Specifically in this case, I find I have a kneejerk reaction when I see the word "spooky" here that makes me feel defensive and want to oppose the change. I know this is silly and I am suppressing it, but it takes effort. In general, avoiding a label that feels pejorative when proposing changes will help people feel good about them.)